frontend/src/data/Dialect.ts

```diff
@@ -37,6 +36,52 @@ export default class Dialect {
     return fromSerialized(await response.text());
   }
 
+  async fetchPrevReportImplementations(baseURI: URI) {
+    const prevVersionUrl = baseURI
+      .clone()
+      .directory("previous")
+      .filename(this.path)
+      .suffix("json")
+      .href();
+
+    let prevReportLines: string | null = null;
+    await fetch(prevVersionUrl)
+      .then(async (response) => {
+        prevReportLines = await this.fetchResponseStream(response, 1);
+      })
+      .catch(() => {
+        //  If the old report isnt available
+        prevReportLines = null;
+      });
+    if (prevReportLines) {
+      const prevReportMetaData = JSON.parse(prevReportLines) as RunInfo;
+      return prevReportMetaData.implementations;
+    }
+    return null;
+  }
+
+  // Helper function to stream and fetch the first `num_lines`
+  async fetchResponseStream(response: Response, num_lines: number) {
```

Comment:
I had done the same however @Julian suggested that it would be better to stream the response and fetch the first line only

Reply:
> Does it really helps so much performance-wise? Couldn't you just read it all to string and then get the 1st line?

Mostly I was surprised this isn't trivial (which I didn't know at the time when suggesting it) -- in Python this is a 2 second thing to do -- in retrospect I wouldn't have suggested it certainly, knowing that it appears it's not in JS, but it seemed unnecessary to tell you to *back it out* after you figured it out.

Reply:
I can roll it back to just fetching the first line from the whole response. I also think that the code is getting complicated as compared to the small time advantage that it would provide.

Reply:
Ideally the server could just return exactly what is needed but It's probably even more troublesome with gh static pages :D